marlowe/src/Language/Marlowe/SemanticsTypes.hs

```diff
@@ -662,16 +708,12 @@ instance Eq Contract where
     When cases1 timeout1 cont1 == When cases2 timeout2 cont2 =
         timeout1 == timeout2 && cont1 == cont2
         && length cases1 == length cases2
-        && let cases = zip cases1 cases2
-               checkCase (Case action1 cont1, Case action2 cont2) =
-                    action1 == action2 && cont1 == cont2
-           in all checkCase cases
+        && all id (zipWith (==) cases1 cases2)
```

Comment:
`all id` = `and`

Reply:
Oh! Thanks. I was looking for a function that did that in Hoogle and couldn't find it for some reason